### PR TITLE
Validate database names for SQLite handler

### LINKED ISSUE #2749: Validate database and collection names
### What should be done?

The `sqlite` handler should validate that database and collection names are valid according to general FerretDB rules.

The `sqlite` backend should implement additional validation for SQLite-specific rules.

We also should change `CollectionToTable` method. Currently, it does two things:
* converts collection name to valid table name on collection creation;
* converts existing collection name to table name.

The second thing should be done by querying the metadata table.

The manglign scheme should be improved too.

### Where?

* https://github.com/FerretDB/FerretDB/tree/main/internal/backends/sqlite
* https://github.com/FerretDB/FerretDB/tree/main/internal/handlers/sqlite

### Definition of Done

TBD
- `sqlite` handler updated;
- unit tests added/updated;
- integration/compatibility tests checked;
- spot refactorings done;

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #2749.

```
[user@nixpc:~/FerretDB]$ task test-integration-sqlite TEST_RUN='TestDatabaseName'
task: [test-integration-sqlite] go test -count=1 -run='TestDatabaseName' -timeout=35m -race=true -tags=ferretdb_debug,ferretdb_tigris,ferretdb_hana -shuffle=on -coverpkg=../... -coverprofile=integration-sqlite.txt . -target-backend=ferretdb-sqlite -target-tls -compat-url='mongodb://username:password@127.0.0.1:47018/?tls=true&tlsCertificateKeyFile=../build/certs/client.pem&tlsCaFile=../build/certs/rootCA-cert.pem' -disable-filter-pushdown

ok      github.com/FerretDB/FerretDB/integration        0.820s  coverage: 40.0% of statements in ../...
```

## Readiness checklist

- [ ] I added/updated unit tests.
- [x] I added/updated integration/compatibility tests.
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/sqlite/msg_create.go [hunk 1]
```diff
@@ -27,6 +28,9 @@ import (
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
+// validateDatabaseNameRe validates FerretDB database name.
+var validateDatabaseNameRe = regexp.MustCompile("^[a-zA-Z_-][a-zA-Z0-9_-]{0,62}$")
+
```
Comment (noisersup) [inline internal/handlers/sqlite/msg_create.go:33]: There's no need to additional validation for SQLite to satisfy validation that we have on other handlers

**Review by w84thesun — APPROVED** [review]
👍

File: integration/basic_test.go [hunk 2]
```diff
@@ -447,6 +447,43 @@ func TestDatabaseName(t *testing.T) {
 
 	t.Parallel()
 
+	t.Run("SpecialCharacters", func(t *testing.T) {
+		ctx, collection := setup.Setup(t)
+		for name, tc := range map[string]struct {
+			db string // database name, defaults to empty string
+
+			err        *mongo.CommandError // required, expected error from MongoDB
```
Comment (chilagrow) [inline integration/basic_test.go:455]: It's not required, also you can remove altMessage

File: integration/basic_test.go [hunk 3]
```diff
@@ -447,6 +447,43 @@ func TestDatabaseName(t *testing.T) {
...
+		}{
+			"Dash": {
+				db: "--",
+			},
+			"Underscore": {
+				db: "__",
+			},
+			"Sqlite": {
+				db: "sqlite_",
+			},
```
Comment (chilagrow) [inline integration/basic_test.go:467]: Could you merge `63ok` test case around L587 to this table test please :pray:

File: integration/basic_test.go [hunk 4]
```diff
@@ -472,6 +509,25 @@ func TestDatabaseName(t *testing.T) {
...
+
+			"WithABackslash": {
+				db: "\\",
+				err: &mongo.CommandError{
+					Name:    "InvalidNamespace",
+					Code:    73,
+					Message: `Invalid namespace specified '\.TestDatabaseName-Err'`,
+				},
+				altMessage: `Invalid namespace: \.TestDatabaseName-Err`,
+			},
```
Comment (chilagrow) [inline integration/basic_test.go:530]: Could you test database name with number prefix please. Something like `0` is great :hugs:

File: internal/handlers/sqlite/msg_create.go [hunk 5]
```diff
@@ -78,6 +82,11 @@ func (h *Handler) MsgCreate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		return nil, err
 	}
 
+	if !validateDatabaseNameRe.MatchString(dbName) {
+		msg := fmt.Sprintf("Invalid namespace: %s.%s", dbName, collectionName)
+		return nil, commonerrors.NewCommandErrorMsg(commonerrors.ErrInvalidNamespace, msg)
```
Comment (chilagrow) [inline internal/handlers/sqlite/msg_create.go:87]: Please use `NewCommandErrorMsgWithArgument`

**Review by chilagrow — COMMENTED** [review]
Just few things :hugs:

File: integration/basic_test.go (hunk 4 above)
Comment (noisersup) [reply integration/basic_test.go:530]: As we don't support number prefixes for postgres, the test will fail for pg.

**Review by w84thesun — COMMENTED** [review]
Please check `TestDatabaseName/Empty` - the error type needs to be changed.

**Comment by noisersup** [comment]
> Please check TestDatabaseName/Empty - the error type needs to be changed.

That's an accident after resolving merge conflicts, the Empty test should be removed.

**Review by w84thesun — APPROVED** [review]
👍

**Review by rumyantseva — APPROVED** [review]
Looks unified!

File: integration/basic_test.go (hunk 3 above)
Comment (AlekSi) [reply integration/basic_test.go:463]: That was done, resolving

**Review by AlekSi — APPROVED** [review]

**Pull request merged**